vllm/v1/request.py

```diff
@@ -70,6 +82,9 @@ def __init__(
         self.output_token_ids = ConstantList(self._output_token_ids)
         self.all_token_ids = ConstantList(self._all_token_ids)
 
+        # grammar objects
+        self.grammar: Optional[Grammar[Any] | Future[Grammar[Any]]] = grammar
```

Comment:
Ditto about ` | ` support, so we should use `Union`

Reply:
sg I will update this accordingly.